src/main/java/com/mongodb/jdbc/MongoSQLResultSet.java

```diff
@@ -1,25 +1,32 @@
 package com.mongodb.jdbc;
 
 import com.mongodb.client.MongoCursor;
+import java.math.BigDecimal;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.SQLFeatureNotSupportedException;
 import java.sql.Statement;
+import org.bson.BsonDocument;
 import org.bson.BsonValue;
 
-public class MongoSQLResultSet extends MongoResultSet implements ResultSet {
-
-    public MongoSQLResultSet(
-            Statement statement, MongoCursor<MongoResultDoc> cursor, boolean relaxed) {
-        super(statement, cursor, relaxed);
+public class MongoSQLResultSet extends MongoResultSet<BsonDocument> implements ResultSet {
+    public MongoSQLResultSet(Statement statement, MongoCursor<BsonDocument> cursor) {
+        super(statement);
+        this.cursor = cursor;
     }
 
     // This is only used for testing, and that is why it has package level access, and the
     // tests have been moved into this package.
-    MongoResultDoc getCurrent() {
+    BsonDocument getCurrent() {
         return current;
     }
 
+    @Override
+    protected boolean checkNull(BsonValue o) {
+        // TODO, actually check if null
```

Comment:
let's correctly implement this check before merging! alternatively, if we want to leave it for a future ticket, let's throw a TODO exception like we're doing elsewhere instead of returning a hard-coded value

Reply:
Exception sounds good, this is out of scope for this ticket.